File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> mp;
        int n = nums.size();
        for(int i = 0; i < n; i++)
        {
            mp[nums[i]]++;
            
        }

        int check = n / 2;
        for(auto it : mp)
        {
            if(it.second > check)
            {
                return it.first;
            }
        }
        return -1;
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap <Integer,Integer> mp = new HashMap<>();
        int n = nums.length;
        for(int i = 0; i < n; i++)
        {
            mp.computeIfPresent(nums[i],(k,v) -> v+1);
            mp.computeIfAbsent(nums[i],(k) -> 1);
        }
        for (Integer num : mp.keySet())
        {
            if(mp.get(num) > n/2)
            {
                return num;
            }

        }
        return -1;
    }
}